app/controllers/users.js

```diff
@@ -168,27 +167,35 @@ exports.user = function (req, res, next, id) {
     });
 };
 
-exports.signupWithEmail = function (req, res) {
-  // get the user credentials from form  req.body.password
-  // req.body.email
-  User
-    .findOne({ email: req.body.email })
-    .then((existingUser, err) => {
-      if (err) throw err;
-      if (existingUser) {
-        return res.json({ message: 'A user with this email address already exists' });
+exports.signUp = (req, res) => {
+  if (req.body.name && req.body.password && req.body.email) {
```

Comment:
use the worst case scenario in if statements to avoid complicated if else statements. Best approach will be

```js
if(no name, no password, no email) {
      return response to user saying missing required arguments.
}

// proceed with creating user

```